cli/tests/test_git_push.rs

```diff
@@ -1382,89 +1146,53 @@ fn test_git_push_changes_with_name_untracked_or_forgotten(subprocess: bool) {
     let output = work_dir.run_jj(["git", "push", "--named", "b1=@-"]);
-    if subprocess {
-        insta::assert_snapshot!(output, @r"
-        ------- stderr -------
-        Changes to push to origin:
-          Add bookmark b1 to 57ec90f54125
-        Error: Failed to push some bookmarks
-        Hint: The following references unexpectedly moved on the remote:
-          refs/heads/b1 (reason: stale info)
-        Hint: Try fetching from the remote, then make the bookmark point to where you want it to be, and push again.
-        [EOF]
-        [exit status: 1]
-        ");
-    } else {
-        // For libgit2, the error is the same but missing the "reason: stale info"
-        insta::assert_snapshot!(output, @r"
-        ------- stderr -------
-        Changes to push to origin:
-          Add bookmark b1 to 57ec90f54125
-        Error: Failed to push some bookmarks
-        Hint: The following references unexpectedly moved on the remote:
-          refs/heads/b1
-        Hint: Try fetching from the remote, then make the bookmark point to where you want it to be, and push again.
-        [EOF]
-        [exit status: 1]
-        ");
-    }
+    insta::assert_snapshot!(output, @r"
+    ------- stderr -------
+    Changes to push to origin:
+      Add bookmark b1 to 57ec90f54125
+    Error: Failed to push some bookmarks
+    Hint: The following references unexpectedly moved on the remote:
+      refs/heads/b1 (reason: stale info)
+    Hint: Try fetching from the remote, then make the bookmark point to where you want it to be, and push again.
+    [EOF]
+    [exit status: 1]
+    ");
 
     // The bookmark is still forgotten
     let output = work_dir.run_jj(["bookmark", "list", "--all", "b1"]);
-    insta::allow_duplicates! {
     insta::assert_snapshot!(output, @"");
-    }
-    // This command behaves differently. In libgit2 flow, a conflict between a
-    // revision and its descendant is resolved in favor of the latter, like in `jj
-    // git fetch`. In subprocess flow, this logic is not implemented. As far as `jj
-    // git push --named` goes, the stricter behavior of the subprocess flow is
-    // probably slightly better.
     let output = work_dir.run_jj(["git", "push", "--named", "b1=@+"]);
-    if subprocess {
-        insta::assert_snapshot!(output, @r"
-        ------- stderr -------
-        Changes to push to origin:
-          Add bookmark b1 to 08fcdf4055ae
-        Error: Failed to push some bookmarks
-        Hint: The following references unexpectedly moved on the remote:
-          refs/heads/b1 (reason: stale info)
-        Hint: Try fetching from the remote, then make the bookmark point to where you want it to be, and push again.
-        [EOF]
-        [exit status: 1]
-        ");
-        // In this case, pushing the bookmark to the same location where it already is
-        // succeeds. TODO: This seems pretty safe, but perhaps it should still show
-        // an error or some sort of warning?
```

Comment:
@ilyagr I think you had opinions about keeping sentiment from the comments here, so feel free to suggest how I should reword this.